File: backend/Intex2.API/Intex2.API/Data/Donation.cs
using System;
using System.Collections.Generic;

namespace Intex2.API.Data;

public partial class Donation
{
    public int? DonationId { get; set; }

    public int? SupporterId { get; set; }

    public string? DonationType { get; set; }

    public string? DonationDate { get; set; }

    public string? IsRecurring { get; set; }

    public string? CampaignName { get; set; }

    public string? ChannelSource { get; set; }

    public string? CurrencyCode { get; set; }

    public string? Amount { get; set; }

    public double? EstimatedValue { get; set; }

    public string? ImpactUnit { get; set; }

    public string? Notes { get; set; }

    public string? ReferralPostId { get; set; }
}

File: frontend/src/types/donation.ts
export interface DonationDto {
  donationId: number | null;
  supporterId: number | null;
  donationType: string | null;
  donationDate: string | null;
  isRecurring: string | null;
  campaignName: string | null;
  channelSource: string | null;
  currencyCode: string | null;
  amount: string | null;
  estimatedValue: number | null;
  impactUnit: string | null;
  notes: string | null;
  referralPostId: string | null;
}

export interface DonationCreate {
  supporterId?: number;
  donationType?: string;
  donationDate?: string;
  isRecurring?: string;
  campaignName?: string;
  channelSource?: string;
  currencyCode?: string;
  amount?: string;
  estimatedValue?: number;
  impactUnit?: string;
  notes?: string;
}

export interface DonationFilters {
  supporterId?: number;
  donationType?: string;
  campaignName?: string;
}

export interface DonationAllocationDto {
  allocationId: number | null;
  donationId: number | null;
  safehouseId: number | null;
  programArea: string | null;
  amountAllocated: number | null;
  allocationDate: string | null;
  allocationNotes: string | null;
}

export interface DonationAllocationCreate {
  donationId?: number;
  safehouseId?: number;
  programArea?: string;
  amountAllocated?: number;
  allocationDate?: string;
  allocationNotes?: string;
}

export interface AllocationFilters {
  donationId?: number;
  safehouseId?: number;
  programArea?: string;
}
